src/Elliptic/BoundaryConditions/ApplyBoundaryCondition.hpp

```diff
@@ -12,11 +12,31 @@
 #include "Domain/Tags/Faces.hpp"
 #include "Elliptic/BoundaryConditions/BoundaryCondition.hpp"
 #include "Elliptic/Utilities/ApplyAt.hpp"
+#include "Parallel/Tags/Metavariables.hpp"
 #include "Utilities/FakeVirtual.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/TMPL.hpp"
 
 namespace elliptic {
+
+namespace detail {
+// Return the `BoundaryConditionClasses`, or get the list of derived classes
+// from `Metavariables::factory_creation` if `BoundaryConditionClasses` is
+// empty.
+template <typename Base, typename BoundaryConditionClasses, typename DbTagsList>
+struct GetBoundaryConditionClasses {
```

Comment:
I'm not quite sure what this is for, but my guess based on the rest of the patch is that it is to allow different boundary conditions on the linear and non-linear solves.  Would templating the base class on the intended use and then specifying both lists in the metavariables work?  I don't know where the alternate list comes from, so it might not.

Reply:
The main use case for this option to bypass the metavars is for the `MinusLaplacian` subdomain preconditioner. It's in the diff of this PR. It uses a Poisson system with a particular boundary condition to approximate the equations, so I pass those boundary conditions down to the operator. I think it works quite well, and it also comes in handy in the tests 🤷‍♂️

Reply:
OK, I think I see.  Is the following correct?

This feature is used by `MinusLaplacian`, which always uses a `Robin` boundary condition.  That condition class is stored in the `MinusLaplacian` class.  It doesn't depend on anything, so it isn't checkpointed and is just constructed whenever `MinusLaplacian` is.  When `MinusLaplacian` calls into the code to apply the boundary condition, it passes that in and, as a template parameter, the information that the class is guaranteed to be `Robin`, not whatever the main solve is using.

If so, that seems fine.  (And use in tests is also fine.)

Reply:
That's exactly right